Thinktecture.Relay.OnPremiseConnector/SignalR/RelayServerConnection.cs

```diff
@@ -126,10 +125,10 @@ public void RegisterOnPremiseTarget(string key, Uri baseUri)
                 await Start();
                 _logger.Info("Connected to relay server. #" + _id);
             }
-            catch (Exception ex)
+            catch (Exception)
             {
                 _logger.Info("***** ERROR WHILE CONNECTING: #" + _id);
-                Delay(5000).ContinueWith(_ => Start());
+                await Delay(5000).ContinueWith(_ => Start());
```

Comment:
We are inside of `async` method `private async Task<TokenResponse> GetAuthorizationTokenAsync()`. Thus `await` seems me do correct. See [here](https://msdn.microsoft.com/en-us/library/hh194873(v=vs.110).aspx) examples from MSDN which do the same. The fact that the code is inside of exception handle should change nothing.